packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/TableCardFormItems/TableCardFormContent.jsx

```diff
@@ -230,8 +230,18 @@ const TableCardFormContent = ({
   // need to handle thresholds from the DataSeriesFormItemModal and convert it to the right format
   const handleDataItemModalChanges = useCallback(
     (card) => {
-      // Consider existing thresholds outside columns
-      const allThresholds = thresholds?.map((threshold) => ({ ...threshold })) || [];
+      // Get datasource from columns with thresholds
+      const columnThresholdDataSource =
+        card?.content?.columns
+          ?.filter((column) => column?.thresholds?.length)
+          ?.map((column) => column.dataSourceId) || [];
+
+      // Consider existing thresholds outside columns and filter existing thresholds from edited column
+      const allThresholds =
+        thresholds
+          ?.filter(({ dataSourceId }) => !columnThresholdDataSource.includes(dataSourceId))
+          ?.map((threshold) => ({ ...threshold })) || [];
+
```

Comment:
dataSourceId is not unique. because you can create threshold with same dataSourceId and different comparison (<, > = ).

Reply:
@herleraja it is considered in the code. Existing records in thresholds(content.thresholds) from the same datasourceid are going to be replaced by the new records(colum.thresholds), which contains all current thresholds.

The code above discards thresholds from the column that is being edited (using datasourceid) and re-insert the set of thresholds which is updated to last change.